InvenTree/report/api.py

```diff
@@ -683,13 +684,36 @@ def get(self, request, *args, **kwargs):
         return self.print(request, orders)
 
 
+class ReportMetadata(RetrieveUpdateAPI):
+    """API endpoint for viewing / updating Report metadata."""
+
+    def get_serializer(self, *args, **kwargs):
+        """Return correct MetadataSerializer instance depending on which model is requested"""
+        # Get type of report, make sure its one of the allowed values
+        reporttype = self.kwargs.get('reporttype', 'PurchaseOrder')
+        assert reporttype in ['PurchaseOrder', 'SalesOrder', 'Build', 'BillOfMaterials', 'Test']
```

Comment:
Please change: asserts are not garunteed to be followed in python, please raise explicitly if the condition is not met.